include/solvers/vof.h

```diff
@@ -422,6 +417,37 @@ class VolumeOfFluid
   void
   assemble_mass_matrix_diagonal(TrilinosWrappers::SparseMatrix &mass_matrix);
 
+  /**
+   * @brief Modification of the solution
+   *
+   * @param i_bc peeling-wetting boundary index
+   *
+   * @param current_solution_cfd current solution for the fluid dynamics
+   */
+  template <typename VectorType>
+  void
+  apply_peeling_wetting(const unsigned int i_bc,
+                        const VectorType & current_solution_cfd);
+
+  /**
+   * @brief Change cell phase, small method called to avoid code repetition and reduce sloppy
+   * error likelihood in apply_peeling_wetting.
+   *
+   * @param type a string stating the needed change ("w" or "p")
```

Comment:
Why use a string to indicate a status? It would be more appropriate to use an enum class. String compares are quite expensive (and prone to mistakes).

Reply:
you are very right, at the time I implemented it I had no better idea. I will look into that!